lib/packages/docutils/rstast.nim

```diff
@@ -97,6 +97,7 @@ type
     anchor*: string           ## anchor, internal link target
                               ## (aka HTML id tag, aka Latex label/hypertarget)
     sons*: RstNodeSeq        ## the node's sons
+    anchorGen*: string        ## generated anchors from context
```

Comment:
Is this additional field really needed? When `anchor` is already defined you just use it; when it's not — you generate it. Anchor is always unique for any node.

Reply:
removed anchorGen (and reuse anchor), but added anchorKind which determines the kind of anchor (needed for book-keeping, among other things to make this work:
```nim
if n.anchorKind == raDefault: computeAnchorGen(n)
```
and knowing the kind of the anchor can be useful for other purposes.